src/org/centos/pipeline/PipelineUtils.groovy

```diff
@@ -113,51 +120,251 @@ def checkLastImage(stage) {
-            "compose_rev=${commit}\n"
+            "compose_url=${env.HTTP_BASE}/artifacts/${env.RSYNC_DIR}/${env.branch}/ostree\n"
+            "compose_rev=${env.commit}\n"
     } else if (messageType == 'image.running') {
             messageProperties = messageProperties +
-                "compose_url=${HTTP_BASE}/artifacts/${RSYNC_DIR}/${branch}/ostree\n"
-                "compose_rev=${commit}\n" +
+                "compose_url=${env.HTTP_BASE}/artifacts/${env.RSYNC_DIR}/${env.branch}/ostree\n"
+                "compose_rev=${env.commit}\n" +
                 "image_url=''\n" +
                 "image_name=''\n" +
                 "type=qcow2\n"
     } else if ((messageType == 'image.complete') || (messageType == 'test.smoke.running') ||
             (messageType == 'test.smoke.compelete')) {
         messageProperties = messageProperties +
-                "compose_url=${HTTP_BASE}/artifacts/${RSYNC_DIR}/${branch}/ostree\n"
-                "compose_rev=${commit}\n" +
-                "image_url=${image2boot}\n" +
-                "image_name=${image_name}\n" +
+                "compose_url=${env.HTTP_BASE}/artifacts/${env.RSYNC_DIR}/${env.branch}/ostree\n"
+                "compose_rev=${env.commit}\n" +
+                "image_url=${env.image2boot}\n" +
+                "image_name=${env.image_name}\n" +
                 "type=qcow2\n"
     } else {
         return [ topic, messageProperties, messageContent ]
     }
     return [ topic, messageProperties, messageContent ]
+}
+
+/**
+ * Library to send message
+ *
+ * variables
+ *  msgProps - The message properties
+ *  msgContent - The content of the message
+ */
+def sendMessage(msgProps, msgContent) {
+    sendCIMessage messageContent: msgContent,
+            messageProperties: msgProps,
+            messageType: 'Custom',
+            overrides: [topic: "${topic}"],
+            providerName: "${MSG_PROVIDER}"
+}
+
+/**
+ * Library to parse CI_MESSAGE and inject its key/value pairs as env variables.
+ *
+ */
+def injectFedmsgVars() {
+
+    // Parse the CI_MESSAGE into a Map
+    def ci_data = new JsonSlurper().parseText(env.CI_MESSAGE)
+
+    // If we have a 'commit' key in the CI_MESSAGE, for each key under 'commit', we
+    // * prepend the key name with fed_
+    // * replace any '-' with '_'
+    // * truncate the value for the key at the first '\n' character
+    // * replace any double-quote characters with single-quote characters in the value for the key.
+
+    if (ci_data['commit']) {
+        ci_data.commit.each { key, value ->
+            env."fed_${key.toString().replaceAll('-', '_')}" =
+                    value.toString().split('\n')[0].replaceAll('"', '\'')
+        }
+        if (env.fed_branch == 'master'){
+            env.branch = 'rawhide'
+        } else {
+            env.branch = env.fedbranch
+        }
+    }
+}
+
+/**
+ * Library to set default environmental variables. Performed once at start of Jenkinsfile
+ */
+def setDefaultEnvVars(){
```

Comment:
some of these defaults are specified in the properties() section above...are we duplicating them?

Reply:
@scoheb you make a good point, it looks like the following are defined in both the properties section and in the setDefaultEnvVars() method:
`RSYNC_USER
RSYNC_SERVER
RSYNC_DIR
MAIN_TOPIC
FEDORA_PRINCIPAL
`